abz/acousticbrainz.py

```diff
@@ -164,3 +164,185 @@ def process(path):
 def cleanup():
     if os.path.isfile(config.settings["profile_file"]):
         os.unlink(config.settings["profile_file"])
+        
+def generate_dataset(path):
+    """Process a direcotry containing a datase
+    
+    Args:
+        path: the relative or absolute path of the dataser rootdir
+    """
+    if not os.path.exists(path):
+        sys.exit(path + " does not exists!")
+    if not os.path.isdir(path):
+        sys.exit(path + " is not a directory!")
+    datasetdict = scan_dir(os.path.abspath(path))
+    print(datasetdict)
+    # submit dataset using the Dataset API
+    submit_dataset(datasetdict)
+    
+def scan_dir(path):
+    """Scan a directory to create the dataset dictionary basic structure
+    
+    Args:
+        path: the absolute path of the dataser rootdir
+    Returns:
+        datasetdict: a dictionary with the basic structure of the dataset
+    """
+
+    dataset_name = os.path.basename(os.path.normpath(path))
+    # handles the basic dataset dictionary structure
+    datasetdict = _datasetdict_structure(dataset_name)
+    itemuuid = None
+    
+    for dirName, sudDirNames, fileNames in os.walk(path):
+        
+        for subdir in sudDirNames:
+            tmpdict = {}
+            tmpdict['name'] = subdir
+            tmpdict['description'] = ''
+            tmpdict['recordings'] = []
+
+            # scan each subfolder in the dataset root
+            subdir = os.path.join(path, subdir)
+    
+            for dirName, subDirNames, fileNames in os.walk(subdir):
```

Comment:
You shouldn't need to call `os.walk` multiple times. It will recurse into subfolders itself